apps/api-extractor/src/analyzer/AstReferenceResolver.ts

```diff
@@ -0,0 +1,161 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as ts from 'typescript';
+import * as tsdoc from '@microsoft/tsdoc';
+
+import { AstSymbolTable, AstEntity } from './AstSymbolTable';
+import { AstDeclaration } from './AstDeclaration';
+import { WorkingPackage } from '../collector/WorkingPackage';
+import { AstModule } from './AstModule';
+import { AstImport } from './AstImport';
+
+export type ResultOrError<T> = Error | T;
+
+export class AstReferenceResolver {
+  private readonly _astSymbolTable: AstSymbolTable;
+  private readonly _workingPackage: WorkingPackage;
+
+  public constructor(astSymbolTable: AstSymbolTable, workingPackage: WorkingPackage) {
+    this._astSymbolTable = astSymbolTable;
+    this._workingPackage = workingPackage;
+  }
+
+  public resolve(declarationReference: tsdoc.DocDeclarationReference): ResultOrError<AstDeclaration> {
+    // Is it referring to the working package?
+    if (declarationReference.packageName !== undefined
+      && declarationReference.packageName !== this._workingPackage.name) {
+      return new Error('External package references are not supported');
```

Comment:
NOTE: AE6 allowed you to `@inheritDoc` from declarations belonging to external NPM packages.  This may be very easy to support, depending on how complete our `ExportAnalyzer` data is. (I don't remember offhand, since that code is very complex.)  If it's not super simple, I'm planning to punt it until after AE7 ships.

Reply:
Follow up:  It's not super simple.  It seemed so, because API Extractor already has a bunch of `AstModule` objects floating around containing most of the information we need.  But even something as simple as resolving `{@link lib1#class2}` can encounter multiple modules called `lib1`.  In order to choose the right one, you need to consider which package folder the reference is coming from.

It's not super difficult.  But I'm not going to hold up AE7 over this, since the test validation could be involved.  (The `@link` tag will work fine with API Documenter because it only sees a single `lib1.api.json`.)